From f55e8be356d9c019676c7e87a5b9cfee969c2b4f Mon Sep 17 00:00:00 2001
From: David Benjamin <davidben@google.com>
Date: Fri, 30 Dec 2022 13:38:26 -0500
Subject: [PATCH] Rename CTX_TEST to X509V3_CTX_TEST.

Squat fewer unprefixed macros.

Update-Note: CTX_TEST appears to be unused. If affected, switch to using
X509V3_set_ctx_test instead.

Change-Id: I43b86c0b6f147bbca85b8bc6b43602fc4f6697c1
Reviewed-on: https://boringssl-review.googlesource.com/c/boringssl/+/56108
Reviewed-by: Bob Beck <bbe@google.com>
Commit-Queue: David Benjamin <davidben@google.com>
(cherry picked from commit 65ad925f51d879f8babd22ecb2ca4dc779b2cc24)
---
 crypto/x509v3/v3_akey.c  | 2 +-
 crypto/x509v3/v3_alt.c   | 4 ++--
 crypto/x509v3/v3_skey.c  | 2 +-
 include/openssl/x509v3.h | 5 ++---
 4 files changed, 6 insertions(+), 7 deletions(-)

diff --git a/crypto/x509v3/v3_akey.c b/crypto/x509v3/v3_akey.c
index 76ac834c9..b4093bf15 100644
--- a/crypto/x509v3/v3_akey.c
+++ b/crypto/x509v3/v3_akey.c
@@ -164,7 +164,7 @@ static void *v2i_AUTHORITY_KEYID(const X509V3_EXT_METHOD *method,
   }
 
   if (!ctx || !ctx->issuer_cert) {
-    if (ctx && (ctx->flags == CTX_TEST)) {
+    if (ctx && (ctx->flags == X509V3_CTX_TEST)) {
       return AUTHORITY_KEYID_new();
     }
     OPENSSL_PUT_ERROR(X509V3, X509V3_R_NO_ISSUER_CERTIFICATE);
diff --git a/crypto/x509v3/v3_alt.c b/crypto/x509v3/v3_alt.c
index cec354a77..660ced582 100644
--- a/crypto/x509v3/v3_alt.c
+++ b/crypto/x509v3/v3_alt.c
@@ -302,7 +302,7 @@ err:
 // Append subject altname of issuer to issuer alt name of subject
 
 static int copy_issuer(const X509V3_CTX *ctx, GENERAL_NAMES *gens) {
-  if (ctx && (ctx->flags == CTX_TEST)) {
+  if (ctx && (ctx->flags == X509V3_CTX_TEST)) {
     return 1;
   }
   if (!ctx || !ctx->issuer_cert) {
@@ -382,7 +382,7 @@ static int copy_email(const X509V3_CTX *ctx, GENERAL_NAMES *gens, int move_p) {
   X509_NAME_ENTRY *ne;
   GENERAL_NAME *gen = NULL;
   int i;
-  if (ctx != NULL && ctx->flags == CTX_TEST) {
+  if (ctx != NULL && ctx->flags == X509V3_CTX_TEST) {
     return 1;
   }
   if (!ctx || (!ctx->subject_cert && !ctx->subject_req)) {
diff --git a/crypto/x509v3/v3_skey.c b/crypto/x509v3/v3_skey.c
index 56fdabc13..eac7c1d05 100644
--- a/crypto/x509v3/v3_skey.c
+++ b/crypto/x509v3/v3_skey.c
@@ -113,7 +113,7 @@ static void *s2i_skey_id(const X509V3_EXT_METHOD *method, const X509V3_CTX *ctx,
     return NULL;
   }
 
-  if (ctx && (ctx->flags == CTX_TEST)) {
+  if (ctx && (ctx->flags == X509V3_CTX_TEST)) {
     return oct;
   }
 
diff --git a/include/openssl/x509v3.h b/include/openssl/x509v3.h
index e1f7191c2..fad4dcceb 100644
--- a/include/openssl/x509v3.h
+++ b/include/openssl/x509v3.h
@@ -576,8 +576,7 @@ struct v3_ext_ctx {
   const CONF *db;
 };
 
-// TODO(davidben): Rename this to |X509V3_CTX_TEST|.
-#define CTX_TEST 0x1
+#define X509V3_CTX_TEST 0x1
 
 // X509V3_set_ctx partially initializes |ctx| with the specified objects. Some
 // string formats will reference fields in these objects. Each object may be
@@ -602,7 +601,7 @@ OPENSSL_EXPORT void X509V3_set_ctx(X509V3_CTX *ctx, const X509 *issuer,
 //
 // TODO(davidben): Can we remove this?
 #define X509V3_set_ctx_test(ctx) \
-  X509V3_set_ctx(ctx, NULL, NULL, NULL, NULL, CTX_TEST)
+  X509V3_set_ctx(ctx, NULL, NULL, NULL, NULL, X509V3_CTX_TEST)
 
 // X509V3_set_nconf partially initializes |ctx| with |conf| as the config
 // database. Some string formats will reference sections in |conf|. |conf| may
